tests/rbd/test_rbd_dm_cache.py

```diff
@@ -0,0 +1,157 @@
+"""Test case covered -
+    CEPH-83575581 - Verify the usage of ceph RBD images as
+    dm-cache and dm-write cache from LVM side to enhance cache mechanism.
+
+    Pre-requisites :
+    1. Cluster must be up and running with capacity to create pool
+    2. We need atleast one client node with ceph-common package,
+       conf and keyring files
+
+    Test Case Flow:
+    1. Create RBD based pool and an Image
+    2. get rbd based image disk using rbd map
+    3. Create physical volume for RBD based disk
+    4. Create volume group for RBD based disk
+    5. Create cache disk, meta disk and data disk for the volume group
+    6. make disk as dm-cache and dm-write cache based type of cache specified
+    7. Create Xfs file system on metadata parted disk for file system purpose
+    8. mount some files on cache disk and write some I/O on it.
+"""
+
+from ceph.utils import get_node_by_id
+from tests.rbd.rbd_utils import initial_rbd_config
+from utility.log import Log
+from utility.lvm_utils import lvm_create, pvcreate, vgcreate
+from utility.utils import run_fio, run_mkfs
+
+log = Log(__name__)
+
+
+def test_configure_cache(rbd_obj, client_node, pool_name, image_name, cache_type, **kw):
+    """Method to configure cache mechansim as dm-cache
+    and dm-write cache for RBD image based devices.
+
+    Args:
+        rbd_obj: RBD object
+        client_node: cache client node
+        pool_name: name of the pool
+        image_name: name of the image
+        cache_type: type of cache i.e dm-cache and dm-write cache
+        **kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 for failure.
+
+    """
+    config = kw.get("config")
+    device_names = []
+    lv_sizes = {"data_disk": "4G", "cache_disk": "2G", "meta_disk": "2G"}
+    vg_name = "test_cache_vg"
+    test_folder_path = "/tmp/test"
+    import pdb
+
+    try:
+        pdb.set_trace()
+        device_names.append(rbd_obj.image_map(pool_name, image_name)[:-1])
+        pvcreate(client_node, device_names[0])
+        vgcreate(client_node, vg_name, device_names[0])
+
+        for name, size in lv_sizes.items():
+            lvm_create(client_node, name, vg_name, size)
+
+        command = f"echo 'y' | lvconvert --type cache-pool --poolmetadata {vg_name}/meta_disk {vg_name}/cache_disk"
```

Comment:
Please add a comment on what this command is doing, Also can this be made as a method in lvm_utils? So that it can become reusable?